metaflow/plugins/argo/argo_workflows.py

```diff
@@ -175,6 +175,14 @@ def _sanitize(name):
         # allowed by Metaflow - guaranteeing uniqueness.
         return name.replace("_", "-")
 
+    @staticmethod
+    def _label_hash(name):
+        # Hash a name for use as a Kubernetes label.
+        # Use the maximum allowed 63 characters for the hash to minimize collisions.
+        # Preserve part of the name for legibility purposes.
```

Comment:
after some thought, preserving legibility is probably not a priority for a label, as these are not meant for human consumption to begin with. If we want to store the template name as well, this can be done in the annotations which do not have the same length limitations. I don't see a use case for storing the full template name at this point though